dts/bindings/pwm/nordic,nrf-sw-pwm.yaml

```diff
@@ -0,0 +1,65 @@
+---
+title: nRF SW PWM
+version: 0.1
+
+description: >
+    This binding gives a base representation of the nRFx S/W PWM
+
+properties:
+    compatible:
+      constraint: "nordic,nrf-sw-pwm"
+
+    label:
+      type: string
+      category: required
+      description: Human readable string describing the device (used by Zephyr for API name)
+      generation: define
+
+    ch0-pin:
+      type: int
+      description: Channel 0 pin
+      category: optional
+      generation: define
+
+    ch1-pin:
+      type: int
+      description: Channel 1 pin
+      category: optional
+      generation: define
+
+    ch2-pin:
+      type: int
+      description: Channel 2 pin
+      category: optional
+      generation: define
+
+    timer-instance:
+      type: int
+      description: Timer instance to use for generating the PWM output signals
+      category: required
+      generation: define
+
+    timer-capture-compare-count:
```

Comment:
This should be rather `channel-count`. The PWM user does not need to know how the driver is implemented, and that for each PWM channel it uses one CC register.

Reply:
Changed to channel-count, made a comment that this needs to be one less than timer capture compare reigsters.